pkg/apis/core/types.go

```diff
@@ -4077,11 +4077,21 @@ type ServicePort struct {
 	// The IP protocol for this port.  Supports "TCP", "UDP", and "SCTP".
 	Protocol Protocol
 
-	// The application protocol for this port.
+	// Used as a hint for implementations to
+	// configure the protocol used between the
+	// implementation and the application it exposes.
 	// This field follows standard Kubernetes label syntax.
-	// Un-prefixed names are reserved for IANA standard service names (as per
+	// Valid values are either:
+	//
+	// * Un-prefixed protocol names - reserved for IANA standard service names (as per
 	// RFC-6335 and https://www.iana.org/assignments/service-names).
-	// Non-standard protocols should use prefixed names such as
+	//
+	// * Kubernetes standard names:
+	//   * 'k8s.io/http2' - http2 over cleartext, aka 'h2c'. https://www.rfc-editor.org/rfc/rfc7540
```

Comment:
this is super confusing, http2 uses TLS by default, cleartext is not commonly used,  did we agree on this name or is up to discussion?

Reply:
Is http2 uses TLS by default? Reading the [RFC](https://www.rfc-editor.org/rfc/rfc7540#section-3.1) my understanding is that it negotiates by default to choose h2/h2c.

should not we use https for http2 over tls? and let the client/app decide whether it should upgrade from http to http2 or not ? (https://www.rfc-editor.org/rfc/rfc7540#section-3.3) 

I think istio works as I explained right? @howardjohn

Reply:
> shouldn't we use https for http2 over tls? and let the client/app decide whether it should upgrade from http to http2 or not ? (https://www.rfc-editor.org/rfc/rfc7540#section-3.3)

by the same token, should we use `http` for http2 without TLS and let the client negotiate an upgrade to h2c?

`k8s.io/http2` meaning http2 _without_ TLS is scary ... I could easily see someone using that not realizing they possibly hinted to another app to attempt non-TLS connections

Reply:
Thats a fair argument. I will change it to ‘h2c’ as it was originally discussed.

i did not find the comment chain where I suggested to change to http2. Maybe it was offline discussions